tests/integ/test_tfs.py

```diff
@@ -0,0 +1,75 @@
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import pytest
+
+import sagemaker
+import sagemaker.predictor
+import sagemaker.utils
+from sagemaker.tensorflow.tfs import TFSModel, TFSPredictor
+from tests.integ.timeout import timeout_and_delete_endpoint_by_name
+
+
+@pytest.fixture(scope='module')
+def tfs_predictor(sagemaker_session, tf_full_version):
+    endpoint_name = sagemaker.utils.name_from_base('sagemaker-tfs')
+    model_data = sagemaker_session.upload_data(path='tests/data/tfs-test-model.tar.gz',
+                                               key_prefix='tfs/models')
+    with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session):
+        model = TFSModel(model_data, 'SageMakerRole', framework_version=tf_full_version)
+        predictor = model.deploy(1, 'ml.c5.xlarge', endpoint_name=endpoint_name)
+        yield predictor
+
+
+# @pytest.mark.continuous_testing
+# @pytest.mark.regional_testing
+def test_predict(tfs_predictor):
+    input_data = {'instances': [1.0, 2.0, 5.0]}
+    expected_result = {'predictions': [3.5, 4.0, 5.5]}
+
+    result = tfs_predictor.predict(input_data)
+    assert expected_result == result
+
+
+def test_predict_generic_json(tfs_predictor):
+    input_data = [[1.0, 2.0, 5.0], [1.0, 2.0, 5.0]]
+    expected_result = {'predictions': [[3.5, 4.0, 5.5], [3.5, 4.0, 5.5]]}
+
+    result = tfs_predictor.predict(input_data)
+    assert expected_result == result
+
+
+def test_predict_jsons(tfs_predictor):
+    input_data = '[1.0, 2.0, 5.0]\n[1.0, 2.0, 5.0]'
+    expected_result = {'predictions': [[3.5, 4.0, 5.5], [3.5, 4.0, 5.5]]}
+
+    predictor = sagemaker.RealTimePredictor(tfs_predictor.endpoint,
+                                            tfs_predictor.sagemaker_session, serializer=None,
+                                            deserializer=sagemaker.predictor.json_deserializer,
+                                            content_type='application/json',
+                                            accept='application/json')
+
+    result = predictor.predict(input_data)
+    assert expected_result == result
+
+
+def test_predict_csv(tfs_predictor):
+    input_data = '1.0,2.0,5.0\n1.0,2.0,5.0'
+    expected_result = {'predictions': [[3.5, 4.0, 5.5], [3.5, 4.0, 5.5]]}
+
+    predictor = TFSPredictor(tfs_predictor.endpoint, tfs_predictor.sagemaker_session,
+                             serializer=sagemaker.predictor.csv_serializer)
+
+    result = predictor.predict(input_data)
+    assert expected_result == result
```

Comment:
We are missing test cases for errors and empty requests.

Reply:
- handled in unit tests and in container package tests
- also this is generic predictor / serializer behavior that is well covered by other tests